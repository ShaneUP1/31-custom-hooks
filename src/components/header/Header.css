.Header {
}

.title {
  font-family: cursive;
  text-shadow: 10px 5px 5px lightgray;
  text-align: center;
  padding-top: 20px;
  font-size: 70px;
  margin-top: 0px;
  margin-bottom: 10px;
}

.light {
  background-color: white;
  color: black;
}

.headerbuttons {
  display: flex;
  flex-direction: column;
  justify-content: left;
  align-items: left;
  font-size: 16px;
}

label {
  margin-left: 10px;
  text-shadow: 10px 5px 5px gray;
}

.homelink {
  color: black;
  margin-top: 10px;
  margin-left: 5px;
  text-decoration: none;
  text-align: center;
  padding: 5px;
  border: 1px solid black;
  box-shadow: 10px 5px 5px gray;
}

.dark {
  background-color: black;
  color: white;
}

.linkcontainer {
  display: flex;
}
