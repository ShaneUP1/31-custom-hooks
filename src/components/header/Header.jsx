import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useTheme } from '../state/themeContext';
import styles from './Header.css';

const Header = () => {
  const { theme, setTheme } = useTheme();

  const toggle = ({ target }) => {
    if(target.checked) setTheme('dark');
    else setTheme('light');
  };

  return (
    <div className={`${styles.Header} ${styles[theme]}`}>
      <h1 className={styles.title}>Last Airbender</h1>
      <div className={styles.headerbuttons}>
        <label htmlFor="toggle">Theme Toggler
        <input id="toggle" type="checkbox" onChange={toggle}/>
        </label>
        <Link className={styles.homelink} to={'/'}>Home</Link>
      </div>
    </div>
  );
};



export default Header;
