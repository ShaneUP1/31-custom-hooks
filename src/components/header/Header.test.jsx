import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { ThemeProvider } from '../state/themeContext';
import { MemoryRouter } from 'react-router-dom';

describe('Header component', () => {
  afterEach(() => cleanup());
  it('renders a header', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </ThemeProvider>)
    ;
    expect(asFragment()).toMatchSnapshot();
  });
});