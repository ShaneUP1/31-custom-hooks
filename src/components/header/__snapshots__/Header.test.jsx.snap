// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a header 1`] = `
<DocumentFragment>
  <div
    class="Header light"
  >
    <h1
      class="title"
    >
      Last Airbender
    </h1>
    <span
      class="linkcontainer"
    >
      <div
        class="headerbuttons"
      >
        <label
          for="toggle"
        >
          Theme Toggler
          <input
            id="toggle"
            type="checkbox"
          />
        </label>
        <a
          class="homelink"
          href="/"
        >
          Home
        </a>
      </div>
    </span>
  </div>
</DocumentFragment>
`;
